src/modelio.cs

```diff
@@ -90,6 +90,28 @@ interface MDLAsset : NSCopying
 		[Export ("canExportFileExtension:")]
 		bool CanExportFileExtension (string extension);
 
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("components", ArgumentSemantic.Copy)]
+		IMDLComponent[] Components { get; }
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("setComponent:forProtocol:")]
+		void SetComponent (IMDLComponent component, Protocol protocol);
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("componentConformingToProtocol:")]
+		[return: NullAllowed]
+		IMDLComponent IsComponentConforming (Protocol protocol);
```

Comment:
@spouliot I totally agree with you but I wanted to keep it close to `MDLObject`'s `IsComponentConforming`...

Most of the changes in this PR are about aligning `MDLObject` and `MDLAsset`.